File: WebApi/Models/Routine.cs
public class Routine : IModel
{
    public int Id { get; set; }
    public string Note { get; set; } = string.Empty;
    public int RhythmInWeeks { get; set; } = 0;
    public string TreatmentCategoryName { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public DateTime Date { get; set; } = DateTime.UtcNow;
    public int? HorseId { get; set; }
    public Horse? Horse { get; set; }
}

File: vue-app/src/classes/Routine.ts
import { LocalIDFactory } from "@/helpers/LocalIDFactory";
import { IRoutine } from "@/interfaces/IRoutine";

export class Routine implements IRoutine {
    localID?: string;
    name: string;
    note: string;
    rhythmInWeeks: number;
    treatmentCategoryName: string;
    createdAt: Date;
    updatedAt: Date;
    date: Date;
    horseId: number | undefined;
    horseName: string | undefined;

    constructor(){
        this.localID = LocalIDFactory.createLocalID();
        this.name = "";
        this.note = "";
        this.rhythmInWeeks = 0;
        this.treatmentCategoryName = "";
        this.createdAt = new Date();
        this.updatedAt = new Date();
        this.date = new Date();
        this.horseId = undefined;
        this.horseName = undefined;
    }

}
